### ISSUE TITLE
New extensions and types commands should be executed before anything else

### ISSUE THREAD

**Issue by IvanDrag0** [pgmodeler/pgmodeler#1818]
**Bug description**
When adding a new data type, extension, etc., that is needed for a table/value, pgModeler shows an error if the call is done before the object is created.

**How to reproduce**

Use pgModeler to add a new extension called `uuid-ossp`:

![image](https://github.com/pgmodeler/pgmodeler/assets/24702285/9583be64-ff05-4145-8239-cadabbc0b334)

Create a new table with a `uuid` data type and a default action `uuid_generate_v4 ()`:

![image](https://github.com/pgmodeler/pgmodeler/assets/24702285/a4fb33bc-7726-4046-b08d-55e9f5c5e8c9)

Export the result to an SQL file (or export directly to the database). It should result the following code:

```
CREATE TABLE public."Test" (
	"UUID" uuid NOT NULL DEFAULT uuid_generate_v4 (),	
	CONSTRAINT "Test_pk" PRIMARY KEY ("UUID")
);
-- ddl-end --
ALTER TABLE public."Test" OWNER TO postgres;
-- ddl-end --


CREATE EXTENSION "uuid-ossp"
WITH SCHEMA public
VERSION '1.1';
-- ddl-end --
COMMENT ON EXTENSION "uuid-ossp" IS E'Generate universally unique identifiers (UUIDs)';
-- ddl-end --
```

Executing this SQL would result in the following error since the function is called before the extension that contains the function is installed:

![image](https://github.com/pgmodeler/pgmodeler/assets/24702285/8960d06d-153e-45c0-9629-8fc215497bdb)

However, if you place the install extension code at the top of the SQL file, everything works as expected. Same thing happens with a new data type if the data type is called before it's initialized/added.

**Info about your desktop**
 - OS: Windows
 - Version: 11
 - Window manager:
 - pgModeler version:  1.1-alpha1a
 - Qt version: 6.5

**Comment by hiepxanh** [comment]
I think there is no wrong with this. You have to place the install extension code at the top of the SQL file.
It is waterfall execute code.

**Comment by IvanDrag0** [comment]
I assume that to "fix" this the code generator would just have to change the order in which some of the SQL code is generated and place specific code on top. It already does it with the "CREATE DATABASE" code by putting those statements on top.

**Comment by rkhaotix** [comment]
Your case is one of the few that the model validation doesn't cover (yet)

https://www.pgmodeler.io/support/docs/67-model-validation?v=1.1.0

**Comment by artemklevtsov** [comment]
Is it solved? I faced with same issue on 1.2.2.
